test/functional/lib/tests.cpp

```diff
@@ -2700,14 +2643,41 @@ LoadApiTest()
     const XDP_API_PROVIDER_DISPATCH *ProviderDispatch;
     const XDP_API_PROVIDER_BINDING_CONTEXT *ProviderBindingContext;
 
-    NTSTATUS Status = InitializeApi(&XdpApiClient, &ProviderDispatch, &ProviderBindingContext, &XdpFuncXdpApiClientDispatch, XDP_API_VERSION_1);
+    NTSTATUS Status =
+        XdpOpenApi(
+            XDP_API_VERSION_1, NULL, NULL, &TestDetach, &XdpFuncXdpApiClientDispatch,
+            1000, &XdpApiClient, &ProviderDispatch, &ProviderBindingContext);
+    TEST_NTSTATUS(Status);
+    if (!NT_SUCCESS(Status)) {
+        return;
+    }
+    XdpUnloadApi(&XdpApiClient);
```

Comment:
could you elaborate this?

Reply:
We should use resource wrappers in this test code, so that the cleanup (e.g. XdpUnloadApi) happens automatically.

This is less important now that we're not able to use exceptions, but it is still more reliable than manual coding, e.g. falling through if/else conditions or using goto.

Reply:
Let me do it in later PR when enabling other tests. sounds like lower priority.

Reply:
Let's do it here, since the code is churning. We don't want to leave TODOs behind in PRs.

Reply:
This test is for raw open/close tests as user mode does.
I tried to implement it, but got link error as

```c++
using unique_xdp_api = wistd::unique_ptr<XDP_API_CLIENT, wil::function_deleter<decltype(&XdpUnloadApi), XdpUnloadApi>>;
static unique_xdp_api XdpApiClient;

bool
TestSetup()
{
#ifdef _KERNEL_MODE
    XDP_API_CLIENT* Client = new (std::nothrow) XDP_API_CLIENT();
    if (Client == nullptr) {
        return false;
    }
    XdpApiClient.reset(Client);
    XdpOpenApi(
        XDP_API_VERSION_LATEST, NULL, NULL, &TestDetach, &XdpFuncXdpApiClientDispatch,
        &TimeoutMs, Client, &XdpApi, &XdpApiProviderBindingContext);
#else
}
```

![image](https://github.com/user-attachments/assets/5c1d2016-ab52-48c7-ad4b-b40081950ba5)

Reply:
@mtfriesen @nibanks @nigriMSFT  is it realistic? even msquic doesn't use unique_ptr for msquic for kernel mode.